src/compiler/core.ts

```diff
@@ -109,6 +109,17 @@ namespace ts {
         return undefined;
     }
 
+    /** Works like Array.prototype.find, returning `undefined` if no element satisfying the predicate is found. */
+    export function tryFind<T>(array: T[], predicate: (element: T, index: number) => boolean): T | undefined {

```

Comment:
Hmm, I expected `find` to have the type that `tryFind` has right now. Why does `find` take `callback` and not `predicate`?
I forgot to mention it when I ran into it last week while trying to use `find`.